Sources/Parsing/Parser.swift

```diff
@@ -59,6 +59,7 @@ extension Parser {
   ///
   /// - Parameter input: A nebulous piece of data to be parsed.
   /// - Returns: A more well-structured value parsed from the given input.
+  @_disfavoredOverload
   @inlinable
   public func parse(_ input: Input) rethrows -> Output {
```

Comment:
This is an easy fallback to match on, but should be disfavored for generic contexts. For example where we should automatically insert `End`s into collection parsers.